166214091 admin can add  books

JuwonAbiola opened the pull request:
#### What does this PR do?

Add books

####  Description of Task to be completed?

An Admin should be able to add books

#### How should this be manually tested?

pull this branch
start the server
on Postman and test the add books route 
/api/v1/admin/book

####  Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#166214091](https://www.pivotaltracker.com/n/projects/2350588/stories/166214091)

#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/42170732/59108045-ee8afe00-8931-11e9-8c04-c9c2a3f8c7a8.png)

allebd commented on server/routes/admin.js:
```diff
@@ -1,12 +1,15 @@
 const { addAuthorValidation } = AuthorValidator;
 
 admin.post(`${BASE_URL}/author`, addAuthorValidation(), addAuthor);
+admin.post(`${BASE_URL}/addBook`, addBook);
```
I think the route should be **${BASE_URL}/book** not **${BASE_URL}/addBook**

allebd commented on .eslintrc.json:
```diff
@@ -16,6 +16,10 @@
         "no-shadow": ["error", { "allow": ["req", "res", "err"] }],
+        "linebreak-style": [
+            2,
+            "windows"
```
Instead of adding this, you can update your VSCode to LF on the right bottom side of your editor

allebd commented on server/controllers/BooksController.js:
```diff
@@ -0,0 +1,32 @@
+ * @returns {JSON Object}
+ */
+
+class BooksController {
```
Your controller can be named **BookController** instead of **BooksController**, same as the file name

KvNGCzA commented on server/controllers/BooksController.js:
```diff
@@ -0,0 +1,32 @@
+
+class BooksController {
+  static async addBook(request, response) {
+    const { title, description, isbn, price, yearPublished, stock, authorId } = request.body;
```
What if its an author that doesn't exist yet, he wouldn't have an id, let's use author name instead so that we can create an author if the author does not exist

JuwonAbiola replied:
done

Lundii reviewed (changes requested):
Good job
Here are what I noticed
* The endpoint on your route should be` api/v1/admin/book
* Superadmin should also be allowed to add books`

Ukhu commented on server/middlewares/verifyToken.js:
```diff
@@ -10,12 +10,12 @@ export default (request, response, next) => {
 
   if (token) {
-    jwt.verify(token, process.env.JWT_KEY, async (error, decoded) => {
+    jwt.verify(token, process.env.JWT_KEY, async(error, decoded) => {
```
Do you have eslint installed as a VSCode extension? It should flag it that you should add a space between the async keyword and the open bracket.

henryade commented on server/routes/admin.js:
```diff
@@ -4,13 +4,17 @@ import UserValidator from '../middlewares/UserValidator';
 const { authorValidation } = AuthorValidator;
+
+const { BookValidation } = BookValidator;
 const authorizeAdmin = authorizeUser(['admin']);
```
this should be put in the route directly

henryade commented on server/routes/admin.js:
```diff
@@ -20,5 +24,6 @@ const { createUserValidation, changeStatusValidation } = UserValidator;
 
 admin.post('/admin/user', verifyToken, authorizeUser(['superadmin']), createUserValidation(), createUser);
```
add comments to routes

JuwonAbiola committed:
166214091-story(admin):addBook
create addBook RESTful route for admin
add route for addBook to the server/routes file
[#166214091]
